### ISSUE TITLE
[MJAVADOC-335] In Maven 3 it is impossible to generate several JavaDoc reports at once

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#634]
**[Antonius Ng](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=antonius)** opened **[MJAVADOC-335](https://issues.apache.org/jira/browse/MJAVADOC-335?redirect=false)** and commented

When I try to generate several JavaDoc reports using mvn site(see pom.xml fragment below), only last one is actually created.
It works perfectly in Maven 2.2.1 and looks broken in Maven 3.0.3. I tried to change pom.xml to Maven 3 style (use Maven Site Plugin \<reportPlugins> section instead of \<reporting>) but it didn't help.

Workaround: use Maven 2.2.1 for site generation.

```xml
    <reporting>
        <plugins>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <version>2.8</version>
                <reportSets>

                    <reportSet>
                        <id>report-1</id>
                        <configuration>
                           <destDir>docs1</destDir>
                            ...
                        </configuration>
                        <reports>
                            <report>javadoc</report>
                        </reports>
                    </reportSet>
                    <reportSet>
                        <id>report-2</id>
                        <configuration>
                           <destDir>docs2</destDir>
                           ...
                        </configuration>
                        <reports>
                            <report>javadoc</report>
                        </reports>
                    </reportSet>                    
                </reportSets>
            </plugin>

        </plugins>
    </reporting>
```

---

**Affects:** 2.8

**Issue Links:**
- [MSHARED-271](https://issues.apache.org/jira/browse/MSHARED-271) Multiple reportSets don't work
 (_**"duplicates"**_)

2 votes, 3 watchers

**Comment by jira-importer** [comment]
**[Lance Speelmon](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=lancespeelmon)** commented

FYI - this is another example of configuring the new way and only the last report being generated:

```
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-site-plugin</artifactId>
  <configuration>
    <reportPlugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-project-info-reports-plugin</artifactId>
        <configuration>
          <dependencyDetailsEnabled>false</dependencyDetailsEnabled>
          <dependencyLocationsEnabled>false</dependencyLocationsEnabled>
        </configuration>
      </plugin>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-javadoc-plugin</artifactId>
         <reportSets>
          <reportSet>
            <id>html</id>
            <reports>
              <report>javadoc</report>
            </reports>
          </reportSet>
          <reportSet>
           <id>jaxb</id>
           <configuration>
            <doclet>com.lunatech.doclets.jax.jaxb.JAXBDoclet</doclet>
            <docletArtifacts>
             <docletArtifact>
              <groupId>com.lunatech.jax-doclets</groupId>
              <artifactId>doclets</artifactId>
              <version>0.10.0</version>
             </docletArtifact>
            </docletArtifacts>
            <destDir>jaxbdocs</destDir>
            <reportOutputDirectory>jaxbdocs</reportOutputDirectory>
           </configuration>
           <reports>
            <report>javadoc</report>
           </reports>
          </reportSet>
          <reportSet>
           <configuration>
            <doclet>com.lunatech.doclets.jax.jaxrs.JAXRSDoclet</doclet>
            <docletArtifacts>
             <docletArtifact>
              <groupId>com.lunatech.jax-doclets</groupId>
              <artifactId>doclets</artifactId>
              <version>0.10.0</version>
             </docletArtifact>
            </docletArtifacts>
            <destDir>jaxrsdocs</destDir>
            <reportOutputDirectory>jaxrsdocs</reportOutputDirectory>
           </configuration>
           <reports>
            <report>javadoc</report>
           </reports>
          </reportSet>
          <reportSet>
           <id>jpa</id>
           <configuration>
            <doclet>com.lunatech.doclets.jax.jpa.JPADoclet</doclet>
            <docletArtifacts>
             <docletArtifact>
              <groupId>com.lunatech.jax-doclets</groupId>
              <artifactId>doclets</artifactId>
              <version>0.10.0</version>
             </docletArtifact>
            </docletArtifacts>
            <destDir>jpadocs</destDir>
            <reportOutputDirectory>jpadocs</reportOutputDirectory>
           </configuration>
           <reports>
            <report>javadoc</report>
           </reports>
          </reportSet>
         </reportSets>
      </plugin>
```

**Comment by jira-importer** [comment]
**[Max Schaefer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=awpxw)** commented

I think that it is actually a bug in maven-reporting-exec component (http://jira.codehaus.org/browse/MSHARED-271). It keeps reportsets in a map using the report goal as a key.

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

fixed by MSHARED-271, which will be included in m-site-p 3.3
